### PR TITLE
Fix variable referenced before assignment error

### PR DESCRIPTION
**Pull request by haonanpan**
Fix error:
UnboundLocalError: local variable 'config' referenced before assignment

Signed-off-by: hpan <hpan@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/daemon/kill_started.py [hunk 1]
```diff
@@ -118,5 +118,6 @@ def start_mark(src_file, dest_file):
                 test.fail("Pid file should not reside")
             libvirtd.start()
         if sysconfig:
-            config.restore()
+            if "config" in locals():
+                config.restore()
             libvirtd.restart()
```
Comment (Yingshun) [inline libvirt/tests/src/daemon/kill_started.py:123]: If there's no configuration changes, do you need to restart libvirtd? I wonder if L123 should be updated too..

**Comment by Yingshun** [comment]
@haonanpan Please update test results, thanks!

File: libvirt/tests/src/daemon/kill_started.py [hunk 2]
```diff
@@ -118,5 +118,6 @@ def start_mark(src_file, dest_file):
                 test.fail("Pid file should not reside")
             libvirtd.start()
         if sysconfig:
```
Comment (kylazhang) [inline libvirt/tests/src/daemon/kill_started.py:120]: I think this line is useless here after change, so I think here should be replaced by L121 directly.

Comment (haonanpan) [reply libvirt/tests/src/daemon/kill_started.py:120]: Ok, there are many things need to revise after contact with the feature owner, I will update the PR soon, thanks!

**Comment by haonanpan** [comment]
Test result:

# avocado run --vt-type libvirt --vt-machine-type q35 daemon.kill_started.sigsegv --vt-connect-uri qemu:///system
JOB ID     : 818c263bbb997387f09566ceea44656830478456
JOB LOG    : /root/avocado/job-results/job-2022-02-24T00.06-818c263/job.log
 (1/1) type_specific.io-github-autotest-libvirt.daemon.kill_started.sigsegv: PASS (13.50 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 14.44 s

File: libvirt/tests/src/daemon/kill_started.py [hunk 3]
```diff
@@ -65,10 +67,11 @@ def start_mark(src_file, dest_file):
     try:
         libvirtd.start()
 
-        if sysconfig:
-            config = utils_config.LibvirtdSysConfig()
-            setattr(config, sysconfig.split('=')[0], sysconfig.split('=')[1])
-            libvirtd.restart()
+        if distro.detect().name == "rhel" and (int(distro.detect().version) <= 8 or not libvirt_version.version_compare(8, 0, 0)):
```
Comment (Yingshun) [inline libvirt/tests/src/daemon/kill_started.py:70]: 1. Please limit the line to a maximum of 79 characters.
2. What about other distro? Do you need to skip below lines for other distro? Could you please check it?

File: libvirt/tests/src/daemon/kill_started.py [hunk 4]
```diff
@@ -117,6 +120,7 @@ def start_mark(src_file, dest_file):
                 libvirtd.start()
                 test.fail("Pid file should not reside")
             libvirtd.start()
-        if sysconfig:
-            config.restore()
-            libvirtd.restart()
+        if distro.detect().name == "rhel" and (int(distro.detect().version) <= 8 or not libvirt_version.version_compare(8, 0, 0)):
```
Comment (Yingshun) [inline libvirt/tests/src/daemon/kill_started.py:123]: Same as above

Comment (haonanpan) [reply libvirt/tests/src/daemon/kill_started.py:123]: ok,I'll check it

**Comment by haonanpan** [comment]
@yafu-1

**Comment by Yingshun** [comment]
@haonanpan  Would you please upload test results for daemon.kill_started?

**Comment by haonanpan** [comment]
Latest test result:
#avocado run --vt-type libvirt --vt-machine-type q35 daemon.kill_started.sigsegv --vt-connect-uri qemu:///system
JOB ID     : 70c5c42ebae2e54e1653cdd73153e7d2f82f5f6a
JOB LOG    : /root/avocado/job-results/job-2022-03-02T22.43-70c5c42/job.log
 (1/1) type_specific.io-github-autotest-libvirt.daemon.kill_started.sigsegv: PASS (10.88 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 11.61 s

**Comment by haonanpan** [comment]
OK,done

**Comment by haonanpan** [comment]
> @haonanpan Would you please upload test results for daemon.kill_started?

OK,done

**Review by yafu-1 — APPROVED** [review]
LGTM

**Comment by kylazhang** [comment]
@yafu-1 I think we need to compatible for the old version?

**Comment by haonanpan** [comment]
> @yafu-1 I think we need to compatible for the old version?

We tested on some old versions like "libvirt-6.0.0+rhel8.5", all tests passed, I'm not sure if that is enough.

**Comment by yafu-1** [comment]
> @yafu-1 I think we need to compatible for the old version?

@kylazhang 
Yes, but i think the 'DAEMON_COREFILE_LIMIT=unlimited' set does not affect the  coredump file of libvirtd even for older libvirt version.

**Review by Yingshun — APPROVED** [review]
LGTM

**Review by kylazhang — APPROVED** [review]

**Pull request merged**